src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -351,7 +351,18 @@ protected override void ProcessRecord()
                     HttpResponseMessage response = GetResponse(client, request);
                     response.EnsureSuccessStatusCode();
 
-                    string contentType = ContentHelper.GetContentType(response);
+                    //It is possible for a response to not include Content-Type in the reponse
+                    //header in which case GetContentType will throw a nullReference exception.
+                    //Try to get the ContentType, otherwise catch the resulting exception.
+                    string contentType = null;
+                    try 
+                    { 
+                        contentType = ContentHelper.GetContentType(response);
+                    }
+                    catch (NullReferenceException) 
```

Comment:
NullReferenceException happens only when you are trying to de-reference null object. Exceptions are for exception situations and it's pretty easy to avoid NullReferenceException: we just need to check for `null` in the appropriate place. In this case, in `src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/ContentHelper.CoreClr.cs` GetContentType.
There is a long chain of properties: `response.Content.Headers.ContentType.MediaType`. I don't know from the top of my head, which one is `null`, in this case, but I think you know.
Also that way we really fixing the problem in the implementation and not only on the caller side.